Source/WebKit/Shared/Cocoa/CoreIPCContacts.mm

```diff
@@ -74,6 +74,170 @@
+    if (m_phoneticGivenName)
+        result.get().phoneticGivenName = *m_phoneticGivenName;
+    if (m_phoneticMiddleName)
+        result.get().phoneticMiddleName = *m_phoneticMiddleName;
+    if (m_phoneticFamilyName)
+        result.get().phoneticFamilyName = *m_phoneticFamilyName;
+    if (m_phoneticOrganizationName)
+        result.get().phoneticOrganizationName = *m_phoneticOrganizationName;
+    if (m_note)
+        result.get().note = *m_note;
+
+    if (m_birthday)
+        result.get().birthday = m_birthday->toID().get();
+    if (m_nonGregorianBirthday)
+        result.get().nonGregorianBirthday = m_nonGregorianBirthday->toID().get();
+
+    if (!m_dates.isEmpty()) {
+        auto labeledValues = adoptNS([NSMutableArray new]);
+        for (auto& tuple : m_dates) {
+            RetainPtr<CNLabeledValue> labeledValue = adoptNS([[PAL::getCNLabeledValueClass() alloc] initWithIdentifier:std::get<0>(tuple) label:std::get<1>(tuple) value:std::get<2>(tuple).toID().get()]);
+            [labeledValues addObject:labeledValue.get()];
+        }
+        result.get().dates = labeledValues.get();
+    }
+
+    if (!m_phoneNumbers.isEmpty()) {
+        auto labeledValues = adoptNS([NSMutableArray new]);
+        for (auto& tuple : m_phoneNumbers) {
+            RetainPtr<CNLabeledValue> labeledValue = adoptNS([[PAL::getCNLabeledValueClass() alloc] initWithIdentifier:std::get<0>(tuple) label:std::get<1>(tuple) value:std::get<2>(tuple).toID().get()]);
+            [labeledValues addObject:labeledValue.get()];
+        }
+        result.get().phoneNumbers = labeledValues.get();
+    }
+
+    if (!m_emailAddresses.isEmpty()) {
+        auto labeledValues = adoptNS([NSMutableArray new]);
+        for (auto& tuple : m_emailAddresses) {
+            RetainPtr<CNLabeledValue> labeledValue = adoptNS([[PAL::getCNLabeledValueClass() alloc] initWithIdentifier:std::get<0>(tuple) label:std::get<1>(tuple) value:std::get<2>(tuple)]);
+            [labeledValues addObject:labeledValue.get()];
+        }
+
+        result.get().emailAddresses = labeledValues.get();
+    }
+
+    if (!m_postalAddresses.isEmpty()) {
+        auto labeledValues = adoptNS([NSMutableArray new]);
+        for (auto& tuple : m_postalAddresses) {
+            RetainPtr<CNLabeledValue> labeledValue = adoptNS([[PAL::getCNLabeledValueClass() alloc] initWithIdentifier:std::get<0>(tuple) label:std::get<1>(tuple) value:std::get<2>(tuple).toID().get()]);
+            [labeledValues addObject:labeledValue.get()];
+        }
+        result.get().postalAddresses = labeledValues.get();
+    }
+
+    if (!m_urlAddresses.isEmpty()) {
+        auto labeledValues = adoptNS([NSMutableArray new]);
+        for (auto& tuple : m_urlAddresses) {
+            RetainPtr<CNLabeledValue> labeledValue = adoptNS([[PAL::getCNLabeledValueClass() alloc] initWithIdentifier:std::get<0>(tuple) label:std::get<1>(tuple) value:std::get<2>(tuple)]);
+            [labeledValues addObject:labeledValue.get()];
+        }
```

Comment:
createNSArray can be used to greatly simplify all of these to a one line lambda.